pallets/msa/src/lib.rs

```diff
@@ -586,30 +596,31 @@ impl<T: Config> Pallet<T> {
 		Ok(())
 	}
 
-	/// Disables a key so that it cannot be used again for the associated MSA
+	/// Deletes KeyInfo for a key so that it cannot be used again for the associated MSA
 	/// # Arguments
-	/// * `key` - The key to expire
+	/// * `key` - The key for which key info needs to be removed
 	/// # Returns
 	/// * [`DispatchResult`]
-	///
-	/// # Errors
-	/// * [`Error::<T>::KeyRevoked`]
-	pub fn revoke_key(key: &T::AccountId) -> DispatchResult {
-		KeyInfoOf::<T>::try_mutate(key, |maybe_info| -> DispatchResult {
-			let mut info = maybe_info.take().ok_or(Error::<T>::NoKeyExists)?;
-
-			ensure!(info.expired == T::BlockNumber::default(), Error::<T>::KeyRevoked);
-
-			let current_block = frame_system::Pallet::<T>::block_number();
-
-			info.expired = current_block;
-
-			*maybe_info = Some(info);
+	pub fn delete_key_info(key: &T::AccountId) -> DispatchResult {
+		KeyInfoOf::<T>::remove(key);
+		Ok(())
+	}
 
+	/// Deletes a key associated with a given MSA
+	/// # Arguments
+	/// * `msa_id` - The MSA for which the key needs to be removed
+	/// * `key` - The key to be removed from the MSA
+	/// # Returns
+	/// * [`DispatchResult`]
+	/// # Errors
+	/// * [`Error::<T>::NoKeyExists`] - If the key does not exist in the MSA
+	pub fn delete_key_for_msa(msa_id: MessageSourceId, key: &T::AccountId) -> DispatchResult {
+		<MsaKeysOf<T>>::try_mutate(msa_id, |key_list| {
+			let index = key_list.binary_search(key);
+			ensure!(index.is_ok(), Error::<T>::NoKeyExists);
+			key_list.remove(index.unwrap());
 			Ok(())
-		})?;
```

Comment:
added this function to remove the key from MSAKeysOf when delete is invoked
two things happen
1. Remove the key from msa_id
2. Remove the keyinfo associated with it